activerecord/lib/active_record/connection_adapters/postgresql/oid/range.rb

```diff
@@ -69,13 +69,24 @@ def type_cast_single_for_database(value)
             def extract_bounds(value)
               from, to = value[1..-2].split(",", 2)
               {
-                from:          (from == "" || from == "-infinity") ? infinity(negative: true) : from,
-                to:            (to == "" || to == "infinity") ? infinity : to,
+                from:          (from == "" || from == "-infinity") ? infinity(negative: true) : unquote(from),
+                to:            (to == "" || to == "infinity") ? infinity : unquote(to),
                 exclude_start: value.start_with?("("),
                 exclude_end:   value.end_with?(")")
               }
             end
 
+            def unquote(value)
+              if value[0] == '"' && value[-1] == '"'
+                value[1..-2].tap do |quoted_value|
+                  quoted_value.gsub!('""', '"')
+                  quoted_value.gsub!('\\\\', '\\')
+                end
```

Comment:
`tap` has an extra overhead, just use `unquoted_value = value[1..-2]` instead.

```suggestion
                unquoted_value = value[1..-2]
                unquoted_value.gsub!('""', '"')
                unquoted_value.gsub!('\\\\', '\\')
                unquoted_value
```

Reply:
Also, can you add tests for the quoted two characters? Looks like any test isn't failed if we lose the lines.

Reply:
This is rather tricky to obtain using only Postgres's range types as test cases, as none of the standard ranges' bound types use `"` or `\` in their serialization formats. I've added a test that defines and uses a varchar range in order to allow bound values that require escaping.